pkg/controller/component/kbagent_test.go

```diff
@@ -0,0 +1,330 @@
+		// otherwise if later it needs to create some new resource objects with the same name,
+		// in race conditions, it will find the existence of old objects, resulting failure to
+		// create the new objects.
+		By("clean resources")
+	}
+
+	BeforeEach(func() {
+		cleanEnv()
+	})
+
+	AfterEach(func() {
+		cleanEnv()
+	})
+
+	kbAgentContainer := func() *corev1.Container {
+		for _, c := range synthesizedComp.PodSpec.Containers {
+			if c.Name == kbAgentContainerName {
+				return &c
+			}
+		}
+		return nil
+	}
+
+	kbAgentInitContainer := func() *corev1.Container {
+		for _, c := range synthesizedComp.PodSpec.InitContainers {
+			if c.Name == kbAgentInitContainerName {
+				return &c
+			}
+		}
+		return nil
+	}
+
+	Context("build kb-agent", func() {
+		BeforeEach(func() {
+			synthesizedComp = &SynthesizedComponent{
+				PodSpec: &corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "test-kbagent",
+							Image: "test-kbagent-image",
+						},
+					},
+				},
+				LifecycleActions: &appsv1alpha1.ComponentLifecycleActions{
+					PostProvision: &appsv1alpha1.Action{
+						Exec: &appsv1alpha1.ExecAction{
+							Command: []string{"echo", "hello"},
+						},
+						TimeoutSeconds: 5,
+						RetryPolicy: &appsv1alpha1.RetryPolicy{
+							MaxRetries:    5,
+							RetryInterval: 10,
+						},
+						PreCondition: &[]appsv1alpha1.PreConditionType{appsv1alpha1.ComponentReadyPreConditionType}[0],
+					},
+					RoleProbe: &appsv1alpha1.Probe{
+						Action: appsv1alpha1.Action{
+							Exec: &appsv1alpha1.ExecAction{
+								Command: []string{"echo", "hello"},
+							},
+							TimeoutSeconds: 5,
+						},
+						InitialDelaySeconds: 5,
+						PeriodSeconds:       1,
+						SuccessThreshold:    3,
+						FailureThreshold:    3,
+					},
+				},
+			}
+		})
+
+		It("nil", func() {
+			synthesizedComp.LifecycleActions = nil
+
+			err := buildKBAgentContainer(synthesizedComp)
+			Expect(err).Should(BeNil())
+			Expect(kbAgentContainer()).Should(BeNil())
+		})
+
+		It("port", func() {
+			err := buildKBAgentContainer(synthesizedComp)
+			Expect(err).Should(BeNil())
+
+			c := kbAgentContainer()
+			Expect(c).ShouldNot(BeNil())
+			Expect(c.Ports).Should(HaveLen(1))
+			Expect(c.Ports[0].ContainerPort).Should(Equal(int32(kbAgentDefaultPort)))
+		})
+
+		It("port - in use", func() {
+			synthesizedComp.PodSpec.Containers[0].Ports = []corev1.ContainerPort{
+				{
+					ContainerPort: kbAgentDefaultPort,
+				},
+			}
+			err := buildKBAgentContainer(synthesizedComp)
+			Expect(err).Should(BeNil())
+
+			c := kbAgentContainer()
+			Expect(c).ShouldNot(BeNil())
+			Expect(c.Ports).Should(HaveLen(1))
+			Expect(c.Ports[0].ContainerPort).Should(Equal(int32(kbAgentDefaultPort + 1)))
+		})
+
+		It("startup env", func() {
+			err := buildKBAgentContainer(synthesizedComp)
+			Expect(err).Should(BeNil())
+
+			c := kbAgentContainer()
+			Expect(c).ShouldNot(BeNil())
+			Expect(c.Env).Should(HaveLen(2))
+		})
+
+		PIt("action env", func() {
```

Comment:
Is there a specific issue causing it to be pending here?

Reply:
Enable it.